src/sql/workbench/browser/editor/tableDesigner/tableDesignerInput.ts

```diff
@@ -56,7 +56,11 @@ export class TableDesignerInput extends EditorInput {
 				this._onDidChangeDirty.fire();
 			}
 		}));
-		this._tableIcon = tableInfo.tableIcon.toString() === '' ? TableIcon.Basic : tableInfo.tableIcon as TableIcon;
+		if (!tableInfo.tableIcon || tableInfo.tableIcon.toString() === '') {
+			this._tableIcon = TableIcon.Basic;
+		} else {
+			this._tableIcon = tableInfo.tableIcon.toString() as TableIcon;
```

Comment:
Why are you doing the toString? `tableInfo.tableIcon` and `_tableIcon` are both `TableIcon` types so you shouldn't need to be doing any casting here